### PR TITLE
use  metrics as CDI bean

### PR DESCRIPTION
**Pull request by pavolloffay**
Access metrics storage through CDI bean. 

It is related to https://github.com/hawkular/hawkular-metrics/pull/502 so it should be merged after it.

### CODE REVIEW

**Comment by pavolloffay** [comment]
@Jiri-Kremser could you please review?

File: hawkular-datamining-dist/src/main/java/org/hawkular/datamining/dist/integration/metrics/CDIMetricsClient.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * @author Pavol Loffay
+ */
+@Specializes
+@ApplicationScoped
+public class CDIMetricsClient extends EmptyMetricsClient {
+
+    @javax.annotation.Resource(lookup = "java:global/Hawkular/Metrics")
+    private MetricsService metricsService;
```
Comment (Jiri-Kremser) [inline hawkular-datamining-dist/src/main/java/org/hawkular/datamining/dist/integration/metrics/CDIMetricsClient.java]: so it assumes the metrics are co-located with datamining module, right? Or can this be consumed also remotely?

**Comment by Jiri-Kremser** [comment]
Good job, just a question. Now, that it looks like the accounts will not be used anymore, is this actually still needed?

File: hawkular-datamining-dist/src/main/java/org/hawkular/datamining/dist/integration/metrics/CDIMetricsClient.java (hunk 1 above)
Comment (pavolloffay) [inline hawkular-datamining-dist/src/main/java/org/hawkular/datamining/dist/integration/metrics/CDIMetricsClient.java]: Yes Metrics has to be deployed in the same app server, It is just a CDI bean.

**Comment by pavolloffay** [comment]
Good point, this might be faster than accessing Metrics via REST API. But it also comples these together (which I don't like). On the other hand  Datamining injects Inventory too and  ``dist`` artifact is designed to be deployed inside Hawkular with other components.

Summary how I see it:
-- tightly coupled together
++ faster than REST calls 

@Jiri-Kremser and @pilhuhn what do you think?  Should be this merged or wait until accounts are removed and use its REST API and assume that accounts will be "never used".

**Comment by pavolloffay** [comment]
https://github.com/hawkular/hawkular-metrics/pull/502 got merged. I would like to merge this too if nobody is against.

**Pull request closed without merging**